### ISSUE TITLE
MigraDoc RTF Document does not print Images (png)

### ISSUE THREAD

**Issue by dnehl** [empira/PDFsharp#11]
# Resources
[Issue.zip](https://github.com/empira/PDFsharp/files/11438822/Issue.zip)
![nehl-it-icon](https://github.com/empira/PDFsharp/assets/40561661/6fd58462-b760-4f4b-acfb-6352d6735355)

# Version
I used the latest preview: 
`<PackageReference Include="PDFsharp" Version="6.0.0-preview-2" />
 <PackageReference Include="PDFsharp-MigraDoc" Version="6.0.0-preview-2" />`

# Reporting an Issue Here
## Expected Behavior
the logo, in this case "nehl-it-icon" should be printed in the pdf and in the rtf document, but it only works with PDF

## Actual Behavior
The image is only visible in the pdf document. In Rtf it shows caracters but nothing else.

## Steps to Reproduce the Behavior
Start the Project in the Issue.zip file and check the difference between word and pdf document

**Comment by ThomasHoevel** [comment]
After a little code change, images in RTF work.
Code page 1252 is not UTF8, so remove this:
```
//public class MyEncodingProvider : EncodingProvider
//{
//    public override Encoding GetEncoding(int codepage)
//    {
//        if (codepage == 1252)
//        {
//            // Return your custom encoding object here
//            return Encoding.UTF8;
//        }
//        return null;
//    }

//    public override Encoding GetEncoding(string name)
//    {
//        // Implement this method if you want to support encoding names
//        return null;
//    }
//}

```


Register the encoder for code page 1252 (this should be included in the RTF Renderer soon, but with 6.0.0-preview-2 it must be called explicitly).
```
        //Encoding.RegisterProvider(new MyEncodingProvider());
        System.Text.Encoding.RegisterProvider(System.Text.CodePagesEncodingProvider.Instance);
```

**Comment by dnehl** [comment]
I tried to register the provider in the startup and also before I create the RTF document, but it still did not work. Now the image ins, not there and also not strange symbols. 
Do I have to config the runtime env or the project file in a specific way?

**Comment by ThomasHoevel** [comment]
I do not know about any special requirements. We just use .NET 6 and I simply call `RegisterProvider` early in the code before creating the RTF Renderer.

The issue submission template worked with the changes shown above.

**Comment by dnehl** [comment]
Ok I found the issue, maybe you can help me to solve this as well.
I want to download the image first and add the byte array to the document.
When I create the pdf document everything, but the rtf document does not contain the image.

I am using the following code to download the image and create  a byte array.


```
public static async Task<string> GetImage(string imageName, CancellationToken cancellationToken = default)
    {
        var httpClient = new HttpClient();
        var url = $"{BaseUrl}/nehl-it-icon.png";
        var response = await httpClient.GetAsync(
            url,
            cancellationToken);

        if (!response.IsSuccessStatusCode)
        {
            return string.Empty;
        }

        return LoadImageFromBytes(await response.Content.ReadAsByteArrayAsync(cancellationToken));
    }

    protected static string LoadImageFromBytes(byte[] cData)
    {
        return $"base64:{Convert.ToBase64String(cData)}";
    }

```

````
var imageFilePath = await GetImage("Generali.png");

var image = document.LastSection.Headers.Primary.AddImage(imageFilePath);
````

**Comment by ThomasHoevel** [comment]
The "BASE64" notation does not currently work with the RTF Renderer. This is a different issue we are researching.

At the moment I'd recommend storing the image in a temporary file if you have to download it from the web.

**Comment by dnehl** [comment]
Ok, thanks for the feedback!
Is there an open issue, where I can watch it?

**Comment by ThomasHoevel** [comment]
BASE64 support for RTF Renderer has been fixed.
Fix will be included with version 6.0.0-preview-3 coming soon (May or June 2023 or so).
